### PR TITLE
Update WritingPesterTests.md to reflect recommended use of ShouldBeErrorId

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
Fix https://github.com/PowerShell/PowerShell/issues/4635

### CODE REVIEW

File: docs/testing-guidelines/WritingPesterTests.md [hunk 1]
```diff
@@ -51,24 +51,18 @@ Describe "One is really one" {
 }
 ```
 
-If you are checking for proper errors, do that in a `try/catch`, and then check `FullyQualifiedErrorId`. Checking against `FullyQualifiedErrorId` is recommended because it does not change based on culture as an error message might.
+If you are checking for proper errors, use the `ShouldBeErrorId` helper defined in HelpersCommon.psm1 module which is in your path if you import `build.psm1`.
+Checking against `FullyQualifiedErrorId` is recommended because it does not change based on culture as an error message might.
 
 ```powershell
 ...
-it "Get-Item on a nonexisting file should have error PathNotFound" {
```
Comment (iSazonov) [inline docs/testing-guidelines/WritingPesterTests.md]: Sometimes there are scenarios where we can't use `ShouldBeErrorId`. I think we need to leave this as an alternative.

File: docs/testing-guidelines/WritingPesterTests.md [hunk 2]
```diff
@@ -51,24 +51,18 @@ Describe "One is really one" {
...
-        throw "No Exception!"
-    }
-    catch
-    {
-        $_.FullyQualifiedErrorId | should be "PathNotFound,Microsoft.PowerShell.Commands.GetItemCommand"
-    }
+It "Get-Item on a nonexisting file should have error PathNotFound" {
+    { Get-Item "ThisFileCannotPossiblyExist" -ErrorAction Stop } | ShouldBeErrorId "PathNotFound,Microsoft.PowerShell.Commands.GetItemCommand"
 }
 ```
```
Comment (iSazonov) [inline docs/testing-guidelines/WritingPesterTests.md]: We could still add two sample patterns:
```powershell
It "Get-Item on a nonexisting file should have error PathNotFound" {  
    $exc = { Get-Item "ThisFileCannotPossiblyExist" -ErrorAction Stop } | ShouldBeErrorId "PathNotFound,Microsoft.PowerShell.Commands.GetItemCommand"
    // Analyze the exception in $exc
    $exc.InnerException ...
   } 

It "Get-Item on a nonexisting file should have error PathNotFound" {  
    { Get-Item "ThisFileCannotPossiblyExist" -ErrorAction Stop } | ShouldBeErrorId "PathNotFound,Microsoft.PowerShell.Commands.GetItemCommand"
    // Analyze the exception in $error
    $error[0].InnerException ...
   }  
```

File: docs/testing-guidelines/WritingPesterTests.md (hunk 1 above)
Comment (SteveL-MSFT) [inline docs/testing-guidelines/WritingPesterTests.md]: The only scenario I can think of is inspecting details of a custom error record like invoke-webr quest, I can add sample for that, any others?

Comment (iSazonov) [inline docs/testing-guidelines/WritingPesterTests.md]: [Here](https://github.com/PowerShell/PowerShell/blob/ffd39b2853b68419eb0fd5e34a89b755c98b8022/test/powershell/engine/Remoting/SSHRemotingAPI.Tests.ps1#L63) and [here](https://github.com/PowerShell/PowerShell/blob/02b5f357a20e6dee9f8e60e3adb9025be3c94490/test/powershell/Language/Scripting/HashtableToPSCustomObjectConversion.Tests.ps1#L106) we didn't use the pattern.

Comment (SteveL-MSFT) [inline docs/testing-guidelines/WritingPesterTests.md]: Seems like even those examples could be converted to use ShouldBeErrorId

File: docs/testing-guidelines/WritingPesterTests.md (hunk 2 above)
Comment (SteveL-MSFT) [inline docs/testing-guidelines/WritingPesterTests.md]: I'll add a sample for this

File: docs/testing-guidelines/WritingPesterTests.md (hunk 1 above)
Comment (iSazonov) [inline docs/testing-guidelines/WritingPesterTests.md]: I think yes.
Closed.

File: docs/testing-guidelines/WritingPesterTests.md (hunk 2 above)
Comment (SteveL-MSFT) [inline docs/testing-guidelines/WritingPesterTests.md]: Sample was added, @iSazonov can you review?

**Review by iSazonov — APPROVED** [review]
@SteveL-MSFT  Do we ready to merge?

**Comment by SteveL-MSFT** [comment]
@iSazonov give it a day after approval for further comments from others, then merge if nothing new

**Pull request merged**